### PR TITLE
cmake: Use find_package(VulkanLoader)

### PR DESCRIPTION
**Pull request by juan-lunarg**
Simplifies finding the VulkanLoader

### CODE REVIEW

**Review by charles-lunarg — APPROVED** [review]

File: layer/tests/CMakeLists.txt [hunk 1]
```diff
@@ -49,7 +49,13 @@ function(LayerTest NAME)
                    layer_tests_main.cpp
                    vktestframework.cpp)
     add_dependencies(${TEST_NAME} ProfilesLayer ${TEST_JSON_FILES})
-    target_link_libraries(${TEST_NAME} Vulkan::Headers Vulkan::Vulkan GTest::gtest GTest::gtest_main Vulkan::LayerSettings)
+    target_link_libraries(${TEST_NAME}
+        Vulkan::Headers
+        Vulkan::LayerSettings
+        GTest::gtest
+        GTest::gtest_main
+        $<TARGET_NAME_IF_EXISTS:Vulkan::Loader>
```
Comment (christophe-lunarg) [inline layer/tests/CMakeLists.txt:57]: Why do we need this? The logic is different from before the PR

File: library/test/CMakeLists.txt [hunk 2]
```diff
@@ -2,7 +2,7 @@ set(test_libraries GTest::gtest GTest::gtest_main Vulkan::Headers Vulkan::Profil
 if(ANDROID)
     list(APPEND test_libraries log android dl vulkan)
 else()
-    list(APPEND test_libraries Vulkan::Vulkan)
+    list(APPEND test_libraries Vulkan::Loader)
```
Comment (christophe-lunarg) [inline library/test/CMakeLists.txt:5]: Much clearer!

**Review by christophe-lunarg — APPROVED** [review]

File: layer/tests/CMakeLists.txt (hunk 1 above)
Comment (juan-lunarg) [reply layer/tests/CMakeLists.txt:57]: Didn't realize this function wasn't used on Android. I'll remove the `TARGET_NAME_IF_EXISTS`

Comment (juan-lunarg) [reply layer/tests/CMakeLists.txt]: fixed

**Pull request merged**